src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -948,6 +950,15 @@ public String getMasterBranch() {
         return default_branch;
     }
 
+    /**
+     * Get Repository template was the repository created from.
+     *
+     * @return the repository template
+     */
+    public GHRepository getTemplateRepository() throws CloneNotSupportedException {
+        return (GHRepository) template_repository.clone();
```

Comment:
Also I suspect you will have to inject the root in the repository, unless it's done automatically by Jackson.

In any case, adding a test to test that you can actually execute an operation on the repository after getting it would be better.

Reply:
The clone method was because I've found an spotbug error "EI_EXPOSE_REP: May expose internal representation by returning reference to mutable object", it's for that reason I implements Clonable in this object, and return an object copy.

Reply:
I think you need to add the following annotation to the method instead:
```java
@SuppressFBWarnings(value = { "EI_EXPOSE_REP" }, justification = "Expected")
```

Reply:
> Also I suspect you will have to inject the root in the repository, unless it's done automatically by Jackson.
> 
> In any case, adding a test to test that you can actually execute an operation on the repository after getting it would be better.

Sorry, my English is terrible, and I'm not sure about that comment. 
Is the provided test sufficient?

```java
    /**
     * Test get repository created from a template repository
     *
     * @throws Exception
     *             the exception
     */
    @Test
    public void testGetRepositoryWithTemplateRepositoryInfo() throws Exception {
        GHRepository testRepo = gitHub.getRepositoryById(repo.getId());
        assertThat(testRepo.getTemplateRepository(), notNullValue());
        assertThat(testRepo.getTemplateRepository().getOwnerName(), equalTo("octocat"));
        assertThat(testRepo.getTemplateRepository().isTemplate(), equalTo(true));
    }

```